### PR TITLE
Modernize Bitcoin Fee Calculator code

### PR DESCRIPTION
**Pull request by Milerius**
## Description

- Use constexpr/noexcept constructor to avoid `may throw during initialization storage`
- Make getFeeCalculator const
- Turn the default calculators constexpr
- Use C++17 inline variable for global value and give more readability
- Turn deprecated usage of virtual in an override function
- Use appropriately final when needed.
- Use noexcept when required
- Use `explicit` for single constructor
- Use `nodiscard` for non-void returning functions
- Do not use c-cast as they imply a `const_cast` by the compiler: https://en.cppreference.com/w/cpp/language/explicit_cast

## How to test

```bash
make -C build tests && build/tests/tests tests
```

## Types of changes

Modernize code (use modern c++, improve readability, more safeness)

## Checklist

- [x] Create pull request as draft initially, unless its complete.
- [x] Add/Update tests to cover changes as needed.
- [ ] Update documentation as needed.
- [ ] If there is a related Issue, mention it in the description.

ps: Regarding the format, my IDE have detected the `.clang-format` and applied format based on it

ps 2: I would like an approval status for the CI before undrafting, I've tested only the C++ tests since I've not modified any C API related files

### CODE REVIEW

File: src/Bitcoin/FeeCalculator.h [hunk 1]
```diff
@@ -10,49 +10,63 @@
...
+inline constexpr double gDefaultBytesBase{10};
+inline constexpr double gSegwitBytesPerInput{101.25};
+inline constexpr double gSegwitBytesPerOutput{31};
+inline constexpr double gSegwitBytesBase{gDefaultBytesBase};
 /// Interface for transaction fee calculator.
 class FeeCalculator {
-public:
-    virtual int64_t calculate(int64_t inputs, int64_t outputs, int64_t byteFee) const = 0;
-    virtual int64_t calculateSingleInput(int64_t byteFee) const = 0;
+  public:
```
Comment (catenocrypt) [inline src/Bitcoin/FeeCalculator.h:21]: I prefer 4-space indentation, a 2-space indentation confuses many editors, and they revert to 2-space indentation.
Either two full levels for `public` and the members, or `public` w/o indentation

File: src/Bitcoin/FeeCalculator.h [hunk 2]
```diff
@@ -10,49 +10,63 @@
 
 namespace TW::Bitcoin {
 
+inline constexpr double gDefaultBytesPerInput{148};
+inline constexpr double gDefaultBytesPerOutput{34};
+inline constexpr double gDefaultBytesBase{10};
+inline constexpr double gSegwitBytesPerInput{101.25};
+inline constexpr double gSegwitBytesPerOutput{31};
+inline constexpr double gSegwitBytesBase{gDefaultBytesBase};
```
Comment (catenocrypt) [inline src/Bitcoin/FeeCalculator.h:18]:
```suggestion
inline constexpr double gSegwitBytesBase{gDefaultBytesBase};

```

**Review by catenocrypt — APPROVED** [review]

**Comment by catenocrypt** [comment]
LGTM, thx

File: src/Bitcoin/FeeCalculator.h (hunk 1 above)
Comment (Milerius) [reply src/Bitcoin/FeeCalculator.h:22]: Example with IndentAccessModifiers :

https://user-images.githubusercontent.com/21139416/171010250-12d2060c-ffb8-4acf-b46c-e6c2574e739c.mov

Comment (Milerius) [reply src/Bitcoin/FeeCalculator.h]: > I prefer 4-space indentation, a 2-space indentation confuses many editors, and they revert to 2-space indentation. Either two full levels for `public` and the members, or `public` w/o indentation

Should be fixed now, I've modified the `.clang-format` to respect this specific behaviour, please let me know if that's better now

**Review by hewigovens — APPROVED** [review]

**Comment by SORA-ALONE** [comment]
Milerius:dev_roman #2511

**Pull request merged**